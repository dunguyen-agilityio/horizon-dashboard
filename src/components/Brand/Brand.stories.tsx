import type { Meta, StoryObj } from '@storybook/react';

import Brand from '.';

const meta = {
  title: 'Components/Brand',
  component: Brand,
  tags: ['autodocs'],
} satisfies Meta<typeof Brand>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
