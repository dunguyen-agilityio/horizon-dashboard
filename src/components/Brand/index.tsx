// Components
import Link from 'next/link';
import { Text } from '@/components';

// Fonts
import { Poppins } from 'next/font/google';

// Types
import { TEXT_SIZE } from '@/types/text';

// Utils
import { cn } from '@nextui-org/theme';

const poppins = Poppins({
  subsets: ['latin'],
  weight: ['400', '700'],
});

const Brand = () => (
  <Link href="/">
    <div className="flex gap-1 justify-center py-12 px-6 lg:p-14 lg:pb-12 bg-white dark:bg-indigo">
      <Text
        as="h1"
        size={TEXT_SIZE.xl}
        className={cn(
          poppins.className,
          'leading-8 font-poppins font-extrabold',
        )}
      >
        HORIZON
      </Text>
      <Text
        as="h1"
        size={TEXT_SIZE.xl}
        className={cn(poppins.className, 'leading-8 font-poppins font-normal')}
      >
        FREE
      </Text>
    </div>
  </Link>
);

export default Brand;
