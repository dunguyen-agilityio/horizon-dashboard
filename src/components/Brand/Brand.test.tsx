import { fireEvent, render } from '@testing-library/react';

// Components
import Brand from '.';

describe('Brand component', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Brand />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot', () => {
    const { getByText } = render(<Brand />);

    const brandName = getByText('HORIZON');
    expect(brandName).toBeInTheDocument();
    fireEvent.click(brandName);
  });
});
