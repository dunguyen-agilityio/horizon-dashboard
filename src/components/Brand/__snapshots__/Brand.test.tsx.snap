// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand component Should match snapshot 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="flex gap-1 justify-center py-12 px-6 lg:px-12 lg:pt-14 lg:pb-12 bg-white dark:bg-indigo"
    >
      <h1
        class="text-primary dark:text-white text-xl leading-extra className leading-8 font-poppins font-extrabold"
      >
        HORIZON
      </h1>
      <h1
        class="text-primary dark:text-white text-xl leading-extra className leading-8 font-poppins font-normal"
      >
        FREE
      </h1>
    </div>
  </a>
</div>
`;
